www/js/common/factories/group.factory.js

```diff
@@ -1,123 +1,85 @@
-    updates['/groups/' + newGroupKey] = groupPostData;
-    $rootScope.profile.isLeader = true;
-    updates['/users/' + $rootScope.profile.uid] = $rootScope.profile;
-
-    return firebase.database().ref().update(updates);
+  GroupFactory.fireBase = function(groupCode) {
+    var groupRef = ref.child('groups/' + groupCode);
+    return $firebaseObject(groupRef);
   };
 
-  GroupFactory.addMember = function (groupCode) {
-    var returnValue = $q.defer();
-    ref.child('groups/' + groupCode + '/active').on('value', function(snapshot){
-      if(snapshot.val()){
-        ref.child('users/' + $rootScope.profile.uid + '/activeCode').set(groupCode);
-        var groupRef = ref.child('groups/' + groupCode + '/members/' + $rootScope.profile.uid);
-        $rootScope.profile.activeCode = groupCode;
-
-        returnValue.resolve(groupRef.update($rootScope.profile));
-      } else {
-        returnValue.reject('Inactive Group!');
-      }
-    }, function(err){
-      returnValue.reject(err);
-    });
+  GroupFactory.createGroup = function (groupDetails) {
 
-    return returnValue.promise;
+    return AuthService.getLoggedInUser()
+      .then(function(user) {
+        var newGroupKey = alphanumeric_unique();
+        console.log(user)
+        var groupPostData = {
+          name: groupDetails.name,
+          members: {},
+          timeCreated: Date.now(),
+          mediaUrl: 'http://www.digital-photography-school.com/wp-content/uploads/2011/11/square-format-03.jpg',
+          groupCode: newGroupKey
+        };
+
+        groupPostData.members[user.uid] = true;
+
+        var updates = {};
+        updates['groups/' + newGroupKey] = groupPostData;
+        updates['users/' + user.uid + '/groups/' + newGroupKey] = true;
+
+        return ref.update(updates);
+      });
   };
 
-  GroupFactory.fetchCurrentGroup = function () {
-    var userGroupRef = $q.defer();
-    ref.child('groups/' + $rootScope.profile.activeCode).on('value', function(snapshot){
-          userGroupRef.resolve(snapshot.val());
-        }, function(errorObject){
-          userGroupRef.reject(errorObject.code);
-        });
-      return userGroupRef.promise;
+  GroupFactory.joinGroup = function (groupCode) {
+    return AuthService.getLoggedInUser()
+      .then(function(user) {
+        var updates = {};
+        updates['groups/' + groupCode + '/members/' + user.uid] = false;
+        updates['users/' + user.uid + '/groups/' + groupCode] = true;
+        return ref.update(updates);
+      });
   };
 
-  GroupFactory.fetchMedia = function () {
-    var mediaObjects = $q.defer();
-    ref.child('groups/' + $rootScope.profile.activeCode + '/media').on('value', function(snapshot){
-      mediaObjects.resolve(snapshot.val());
-    }, function(errorObject){
-      mediaObjects.reject(errorObject.code);
-    });
-    return mediaObjects.promise;
+  function returnGroup(groupId) {
+    var group = $q.defer()
+    ref.child('groups/' + groupId).on('value', function(snapshot){
```

Comment:
You might write a utility wrapper for this that could work for any firebase ref and probably even put it in its own factory. Something like `getChildValue(path) => snapshotValue`.